services/src/datasets/external/netcdfcf/overviews.rs

```diff
@@ -467,6 +467,17 @@ pub fn remove_overviews(dataset_path: &Path, overview_path: &Path, force: bool)
 
     fs::remove_dir_all(&out_folder_path).boxed_context(error::CannotRemoveOverviews)?;
 
+    // traverse up and remove folder if empty
```

Comment:
can you not just defer the creation of the folder until you successfully opened the gdal dataset?

Reply:
[74e9b52](https://github.com/geo-engine/geoengine/pull/592/commits/74e9b52580aa2221f3c645b177c20cf5f84d8fff)